README.md

```diff
@@ -145,17 +174,17 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
-
+"java -ea SentenceChecker" to run with assertions :)
+SentenceChecker.java
```

Comment:
Congratulations: this is the first Java program I've seen in a while.

You used a Hashmap, which is great. and you traversed it nicely. The only problem is here:

~~~ java
        for (int i = 0; i < sent.length; i++) {
            temp = new Pair<>(currState, sent[i]);
            if (states.containsKey(temp)) { //checks if pair is valid
                currState = states.get(temp); //sets next state to current state
                if (currState.equals("END")) { //checks if reached end state
                    return true;
                }
            }
~~~

What happens if the hashmap doesn't contain the key? You just ignore the word and look at the next. As a result, the program accepts "the handsome dog ate".

The fix is easy:

~~~ JAVA
        for (int i = 0; i < sent.length; i++) {
            temp = new Pair<>(currState, sent[i]);
            if (states.containsKey(temp)) { //checks if pair is valid
            // ...
            }
            else
                return false;     // << ...........
        }
~~~

# -2